app/policies/pageflow/account_policy.rb

```diff
@@ -0,0 +1,139 @@
+module Pageflow
+  class AccountPolicy
+    class Scope
+      attr_reader :user, :scope
+
+      def initialize(user, scope)
+        @user = user
+        @scope = scope
+      end
+
+      def resolve
+        if user.admin?
+          scope.all
+        else
+          scope.joins(memberships_for_account(user)).where(membership_is_present)
+        end
+      end
+
+      def entry_creatable
+        if user.admin?
+          scope.all
+        else
+          scope.joins(publisher_memberships_for_account(user)).where(membership_is_present)
+        end
+      end
+
+      def entry_movable
+        entry_creatable
+      end
+
+      def themings_accessible
+        entry_creatable
+      end
+
+      def member_addable
+        if user.admin?
+          scope.all
+        else
+          scope.joins(manager_memberships_for_account(user)).where(membership_is_present)
+        end
+      end
+
+      private
+
+      def sanitize_sql_array(array)
+        ActiveRecord::Base.send(:sanitize_sql_array, array)
+      end
+
+      def memberships_for_account(user)
+        sanitize_sql_array(['LEFT OUTER JOIN pageflow_memberships ON ' \
+                            'pageflow_memberships.user_id = :user_id AND ' \
+                            'pageflow_memberships.entity_id = pageflow_accounts.id AND ' \
+                            'pageflow_memberships.entity_type = "Pageflow::Account" AND ' \
+                            'pageflow_memberships.role IN '\
+                            '("member", "previewer", "editor", "publisher", "manager")',
+                            user_id: user.id])
+      end
+
+      def publisher_memberships_for_account(user)
+        sanitize_sql_array(['LEFT OUTER JOIN pageflow_memberships ON ' \
+                            'pageflow_memberships.user_id = :user_id AND ' \
+                            'pageflow_memberships.entity_id = pageflow_accounts.id AND ' \
+                            'pageflow_memberships.entity_type = "Pageflow::Account" AND ' \
+                            'pageflow_memberships.role IN ("publisher", "manager")',
+                            user_id: user.id])
+      end
+
+      def manager_memberships_for_account(user)
+        sanitize_sql_array(['LEFT OUTER JOIN pageflow_memberships ON ' \
+                            'pageflow_memberships.user_id = :user_id AND ' \
+                            'pageflow_memberships.entity_id = pageflow_accounts.id AND ' \
+                            'pageflow_memberships.entity_type = "Pageflow::Account" AND ' \
+                            'pageflow_memberships.role IN ("manager")',
+                            user_id: user.id])
+      end
```

Comment:
These three methods differ only in details. I think a lot of the duplication could be removed by extracting a method.